src/tasks/RuntimeConfigParser/RuntimeConfigParser.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Reflection.Metadata;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public class RuntimeConfigParserTask : Task
+{
+    /// <summary>
+    /// The path to runtimeconfig.json file.
+    /// </summary>
+    [Required]
+    public string RuntimeConfigFile { get; set; } = "";
+
+    /// <summary>
+    /// The path to the output binary file.
+    /// </summary>
+    [Required]
+    public string OutputFile { get; set; } = "";
+
+    /// <summary>
+    /// List of properties reserved for the host.
+    /// </summary>
+    public ITaskItem[] ReservedProperties { get; set; } = Array.Empty<ITaskItem>();
+
+    public override bool Execute()
+    {
+        if (string.IsNullOrEmpty(RuntimeConfigFile))
+        {
+            throw new ArgumentException($"'{nameof(RuntimeConfigFile)}' is required.", nameof(RuntimeConfigFile));
```

Comment:
Generally tasks shouldn't throw unhandled exceptions for bad parameters. They should `Log.LogError(...)` then `return !Log.HasLoggedErrors;`

The build fails either way, but nobody has to see the callstack, and it doesn't look like there's a bug in the task.

Reply:
BTW, I don't think it's possible for non-array task parameters marked `[Required]` to be empty or null - the task would not get invoked - but theres no harm in the validation of course.

Reply:
Fixed. Thanks for the suggestion!